internal/credential/credential.go

```diff
@@ -154,3 +155,11 @@ type SshPrivateKey interface {
 	PrivateKey() PrivateKey
 	PrivateKeyPassphrase() []byte
 }
+
+// SortResource defines a struct used to hold information
+// relating to a resource that is being used for sorting and
+// paginating.
+type SortResource struct {
+	PublicId   string
+	UpdateTime time.Time
+}
```

Comment:
This is used in the new credential domain-wide `WithStartPageAfterItem` option and can be used for a library, store or credential.

Reply:
A [boundary.Resource](https://github.com/hashicorp/boundary/blob/main/internal/boundary/boundary.go#L28-L32) has a `GetPublicId()` and a `GetUpdateTime()` method. Could that interface be used instead of introducing a new type?

Reply:
That's a great idea! Let me try that

Reply:
I really liked this idea but it wasn't quite possible to use `boundary.Resource`, so I defined a new interface in the options file. I also added a function to the refresh token to convert to a "partial resource" which can be used for this. Let me know what you think!